src/Adapter/Redis/RedisCachePool.php

```diff
@@ -29,10 +30,19 @@ class RedisCachePool extends AbstractCachePool implements HierarchicalPoolInterf
     protected $cache;
 
     /**
-     * @param \Redis $cache
+     * @param \Redis|\RedisArray|\RedisCluster $cache
      */
-    public function __construct(\Redis $cache)
+    public function __construct($cache)
     {
+        if (!$cache instanceof \Redis
```

Comment:
shouldn't this check be more like this 

if ($cache instanceof \Redis || $cache instanceof \RedisArray || $cache instanceof \RedisCluster) {
     $this->cache = $cache;
} else {
    throw new CachePoolException('Cache instance must be of type \Redis, \RedisArray or \RedisCluster');
}

Reply:
I can certainly make it like that. But I’d push back a little and say the else clause isn’t needed at all and just muddies the code a bit.  Why execute anything if it doesn’t meet the conditions needed to execute?  However this is not my package, so if you’d like that change I can certainly make it.

Reply:
I can certainly make it like that. But I’d push back a little and say the else clause isn’t needed at all and just muddies the code a bit.  Why execute anything if it doesn’t meet the conditions needed to execute?  However this is not my package, so if you’d like that change I can certainly make it.

Reply:
Good catch... I'll fix this

Reply:
Maybe you can add a small test for this

Reply:
Test was added.  Thanks agaib